kartothek/io/dask/delayed.py

```diff
@@ -582,3 +585,41 @@ def store_delayed_as_dataset(
         metadata_merger=metadata_merger,
         metadata_storage_format=metadata_storage_format,
     )
+
+
+@default_docs
+def build_dataset_indices__delayed(
+    store, dataset_uuid, columns, blocksize=None, factory=None
+):
+    """
+    Function which builds a :class:`~kartothek.core.index.ExplicitSecondaryIndex`.
+
+    This function loads the dataset, computes the requested indices and writes
+    the indices to the dataset. The dataset partitions itself are not mutated.
+
+    Parameters
+    ----------
+
+    Returns
+    -------
+    A dask.delayed computation object.
+    """
+    ds_factory = _ensure_factory(
+        dataset_uuid=dataset_uuid,
+        store=store,
+        factory=factory,
+        load_dataset_metadata=False,
+    )
+
+    mps = list(read_dataset_as_metapartitions__iterator(factory=ds_factory))
```

Comment:
Use `kartothek.io.eager.read_dataset_as_metapartitions` if you want a list of metapartitions